sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricCell.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.metrics;
+
+import java.util.concurrent.atomic.AtomicReference;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
+
+/**
+ * Interface for reporting metric updates of type {@code T} from inside worker harness.
```

Comment:
So `MetricCell` is what the worker harness uses to update its internal state - it just needs `getUpdateIfDirty` and `getCumulative`, since it can never write to the metric. It would be nice to explicitly factor this out.

Then, on the `MetricsContainer` what is actually exposed is the `DistributionCell` types, etc, where the useful methods are actually the additional stuff not contained here.

Can the ownership of the dirty bit be more clearly separately from the ownership of the value?